### ISSUE TITLE
Rule 951260

### ISSUE THREAD

**Issue by azurit** [coreruleset/coreruleset#1912]
### Description

Rule ID 951260 is doing lots of false positives, it is quite easy to trigger it. It has two conditions:
```
SecRule TX:sql_error_match "@eq 1" \
SecRule RESPONSE_BODY "@rx (?i)(?:Sybase message:|Warning.*sybase.*|Sybase.*Server message.*)" \
```
So, if response contains anything from file `sql-errors.data` AND contains at least one phrase which match these regexpes:
```
Sybase message:
Warning.*sybase.*
Sybase.*Server message.*
```
If you look into `sql-errors.data`, for example these phrases are found:
```
sybase
Warning
Sybase message
Sybase message:
Server message
```

So, in lots of cases, the rule acts as if it has only one of the conditions above. I believe one of these should be done:
 - remove rule 951260
 - remove phrases above from `sql-errors.data` (maybe some more)

### Your Environment

* CRS version (e.g., v3.2.0): 3.3.0
* Paranoia level setting: PL1
* ModSecurity version (e.g., 2.9.3): 2.9.3
* Web Server and version (e.g., apache 2.4.41): 2.4
* Operating System and version: Debian Buster

### Confirmation

[x] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by dune73** [comment]
So this rule is based on sql-errors and then reiterates the same patterns in a chained condition? Not a very smart rule.

You are the first to complain, though, and I reckon this is because few sybase installations exist (and ModSec3 on NGINX does not execute the response body rules).

I think the phrases are a bit short / stupid. If `Sybase message:` can trigger an FP, it's too broad.

But can we restrict the patterns a bit more. I mean who does a real Sybase message look like these days? Are there other patterns in said message?

**Comment by azurit** [comment]
@dune73 Sorry, i didn't notice your response. :-( I was VERY busy few last weeks. Unfortunately, i don't run any Sybase database so i cannot tell. That sybase phrase is quite common: for example, it's included in phpinfo() output. I think we should talk about this more at issue meeting. Thank you!

**Comment by lifeforms** [comment]
It seems the sybase rule is a bit broad, such that a long page containing 'Sybase' and 'Warning' will be treated like an error, even if those are miles apart. Sybase is a legacy database that's apparently still in use, from the fact that people are using its PHP extension and having the FP turn up in phpinfo() pages. I'll do an install of the PHP sybase module and try to generate some error strings, so that we can narrow down the pattern. For instance instead of `Warning.*sybase.*` we could do something like `Warning.{1-15}sybase.*` or something (example pulled out of thin air).

**Comment by azurit** [comment]
@lifeforms Are you still working on this? Thanks for info!

**Comment by lifeforms** [comment]
It turns out that the PHP Sybase extension is deprecated. The `sybase_foo()` functions are no longer avaiable in PHP 7.x.

I found an old machine running PHP 5.6 and was able to install the sybase extension there. I could not find a Sybase server to download, it's deprecated. But I can try to produce a warning. It comes in the following format:

```
<b>Warning</b>:  sybase_connect(): Unable to connect to server: localhost in <b>sybase.php</b> on line <b>11</b><br />
```

On the CLI the format is:

```
Warning: sybase_connect(): Unable to connect to server: localhost
```

So we could change the 'warning' pattern to something like `Warning.{2-7}sybase`.

I don't know how to get the other errors in the pattern. I think `Sybase message:` is probably good to keep, this should be rare in output.

Then finally there is `Sybase.*Server message.*` I don't know how to reproduce that error so I can't comment on it. Is that something that gives a lot of FP?

**Comment by dune73** [comment]
@azurit: Do you have an opinion here? If I read you correctly, the FPs you are getting are principally due to `phpinfo()` triggering this. If we could avoid that and maybe also adopt the easier proposals by @lifeforms, would you then be satisfied?

**Comment by azurit** [comment]
> So we could change the 'warning' pattern to something like `Warning.{2-7}sybase`.

I think most of the FPs are generated by pattern `Warning.*sybase.*` so maybe this would be sufficient.

**Comment by lifeforms** [comment]
Something like that would work. Maybe we can be generous and give a bit more chars, like `{2-20}` to be on the safe side.

**Comment by dune73** [comment]
Sounds like a plan to me. Next step PR?

**Comment by dune73** [comment]
Any interest to fix this here?

**Comment by azurit** [comment]
I can do it but it's assigned to @lifeforms .

**Comment by dune73** [comment]
True. But given it's been assigned to him for close to a year now, I am confident he would not mind you taking it out of his hands. :)

He's constantly overworked and too many issues assigned to him.

**Comment by lifeforms** [comment]
It would be awesome if you could do it @azurit , as said I am still swamped and cannot be as active as I would like.
